metapi/metapi.js

```diff
@@ -251,47 +251,94 @@ window.metapi = {};
    */
   metapi.getPost = function (ident, key, options, callback) {
     options = options || {};
+    options.key = key;
+    options.page = options.page || 1;
 
-    var overwrite = options.hasOwnProperty("forceReload") && delete options["forceReload"];
+    var overwrite = options.hasOwnProperty("forceReload") && options["forceReload"];
+    delete options["forceReload"];
 
-    var optionString = "";
-    var optionNames = Object.keys(options);
-    for (var i = 0; i < optionNames.length; i++) {
-      optionString += "&" + optionNames[i] + "=" + options[optionNames[i]];
+    if (!(ident instanceof Array)) {
+      ident = [ident];
     }
 
-    var cached = metapi.postCache.get(ident);
-    if (cached && !overwrite) {
-      return new metapi.Response(true, cached);
-    }
+    // Filter duplicates
+    ident = ident.sort().filter(function (item, pos, ary) {
+      return !pos || item !== ary[pos - 1];
+    });
 
-    var fetchUrl = "";
-    if (typeof ident === "string") {
-      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/urls?urls=" + ident + "&key=" + key + optionString;
-    }
-    else if (typeof ident === "number") {
-      fetchUrl = "https://metasmoke.erwaysoftware.com/api/posts/" + ident + "?key=" + key + optionString;
-    }
+    var response = [];
+    var toLoad = [];
 
-    $.ajax({
-      type: "GET",
-      url: fetchUrl
-    })
-    .done(function (data) {
-      var items = data.items;
-      if (items.length > 0 && items[0]) {
-        metapi.postCache.add(ident, items[0]);
-        callback(new metapi.Response(true, items[0]));
+    for (var j = 0; j < ident.length; j++) {
+      var cached = metapi.postCache.get(ident[j]);
+      if (cached && options.page === 1 && !overwrite) {
+        response.push(new metapi.Response(true, cached));
```

Comment:
This will add a new `metapi.Response` to the response array for each post; ideally, we have one `metapi.Response` containing every post.

*(promise I'm not nitpicking, I'd just like this to be as good as possible :) )*